src/server_manager/web_app/app.ts

```diff
@@ -911,6 +915,28 @@ export class App {
         });
   }
 
+  private setPortForNewAccessKeys(port: number, success: () => void, fail: (message: string) => void) {
+    this.appRoot.showNotification(this.appRoot.localize("notification-keys-port-saving"));
```

Comment:
You could rewrite this function with async/await to avoid nesting.

Reply:
None of the other event listeners use async/await, and I thought it would be negative for readability to have just one that is, as a future reviewer would wonder why this one is different from the others.  It also only saves nesting only on a couple lines, all the error handling would still have the same indentation level inside the catch block.

Reply:
When this code was written, async/await were not yet available. There is currently a mix of promise and async/await in this file. I think we should eventually migate the old code and not introduce more promise-style code in new PRs.